automated-testing/performance-testing/readme.md

```diff
@@ -0,0 +1,96 @@
+# Peformance Testing
+
+Performance Testing is an overloaded term that is used to refer to several sub categories of performance related testing, each of which has different purpose. A good description of overall performance testing is as follows:
+
+"*Performance testing is a type of testing intended to determine the responsiveness, throughput, reliability, and/or scalability of a system under a given workload.* " -  [Performance Testing Guidance for Web Applications](https://docs.microsoft.com/en-us/archive/blogs/dajung/ebook-pnp-performance-testing-guidance-for-web-applications).
+
+Before getting into the different subcategories of performance tests let us understand why is performance testing typically done.
+
+## Why Performance Testing
+
+Performance testing is commonly conducted to accomplish one or more the following:
+
+* To help in accessing whether a **system is ready for Release**:
+  * Estimating / Predicting the performance characteristics (such as response time, failure rate, throughput) which an application is likely to have when it is released in to production. The results can help in predicting the satisfaction level of the users when interacting with the system. The predicted values can also be compared with agreed values (success criteria) for the performance characteristics when available.
+  * To help in accessing the adequacy of the infrastructure / managed service sku's to meet the desired performance characteristics of a system
+  * Identifying bottlenecks and issues with the application at different load levels
+* To compare the **performance impact of application changes**
+  * Comparing the performance characteristics of an application after a change to the values of performance characteristics during previous runs (or baseline values), can provide an indication of performance issues or enhancements introduced due to a change
+* To **support system tuning**
+  * Comparing performance characteristics of a system for different system configurations
+
+## Key Performance Testing categories
+
+* **Performance Testing** : This category is the super set of all sub categories of performance related testing. It validates/determines the speed, scalability or reliability characteristics of the system under test. Performance testing focusses on achieving the response times, throughput, and resource utilization levels which meet the performance objectives of a system
+* **Load Testing** : This is the subcategory of performance testing which focusses on validating the performance characteristics of a system, when the system faces load volumes which are expected during production operation.
+* **Stress Testing** : This is the subcategory of performance testing which focusses on validating the performance characteristics of a system, when the system faces load volumes or conditions beyond those expected during production operations. This could include tests which limit the memory, disk available to the system.
```

Comment:
I somewhat agree with what you have here, but I think it is not clearly distinguished from load testing? For instance, I tend to think of stress testing as pressuring the system to a degree that it would degrade performance, the goal is to evaluate how does the system handle such conditions. Does it recover (i.e. scale-out) or does it just break and fail?

Reply:
@alkofahi I have incorporated the comment